pkg/cheatsheet/generate.go

```diff
@@ -120,131 +133,66 @@ func formatBinding(binding *types.Binding) string {
 	return fmt.Sprintf("  <kbd>%s</kbd>: %s\n", gui.GetKeyDisplay(binding.Key), binding.Description)
 }
 
-func getBindingSections(mApp *app.App) []*bindingSection {
-	bindingSections := []*bindingSection{}
-
-	bindings := mApp.Gui.GetCheatsheetKeybindings()
-
-	type contextAndViewType struct {
-		subtitle string
-		title    string
-	}
-
-	contextAndViewBindingMap := map[contextAndViewType][]*types.Binding{}
-
-outer:
-	for _, binding := range bindings {
-		if binding.Tag == "navigation" {
-			key := contextAndViewType{subtitle: "", title: "navigation"}
-			existing := contextAndViewBindingMap[key]
-			if existing == nil {
-				contextAndViewBindingMap[key] = []*types.Binding{binding}
-			} else {
-				for _, navBinding := range contextAndViewBindingMap[key] {
-					if navBinding.Description == binding.Description {
-						continue outer
-					}
-				}
-				contextAndViewBindingMap[key] = append(contextAndViewBindingMap[key], binding)
-			}
-
-			continue outer
-		}
-
-		contexts := []string{}
-		if len(binding.Contexts) == 0 {
-			contexts = append(contexts, "")
-		} else {
-			contexts = append(contexts, binding.Contexts...)
-		}
-
-		for _, context := range contexts {
-			key := contextAndViewType{subtitle: context, title: binding.ViewName}
-			existing := contextAndViewBindingMap[key]
-			if existing == nil {
-				contextAndViewBindingMap[key] = []*types.Binding{binding}
-			} else {
-				contextAndViewBindingMap[key] = append(contextAndViewBindingMap[key], binding)
-			}
-		}
-	}
-
-	type groupedBindingsType struct {
-		contextAndView contextAndViewType
-		bindings       []*types.Binding
-	}
+func getBindingSections(bindings []*types.Binding, tr *i18n.TranslationSet) []*bindingSection {
```

Comment:
This is the most substantial refactor of this PR (at time of writing). I took a long time to work out what the original code did. I hacked it together without thinking too hard about maintainability given that it's just a script for generating cheatsheets as opposed to being part of lazygit itself, but nonetheless it was not easy to read. The new approach is far more concise, although I've added a MultiGroupBy function to the utils package for grouping the bindings by headers, which is a fairly bespoke usecase that arguably should be inlined (or at least put in a non-generic function).